utils/merkletrie/fsnoder/new.go

```diff
@@ -0,0 +1,193 @@
+package fsnoder
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"gopkg.in/src-d/go-git.v4/utils/merkletrie/hasher"
```

Comment:
You are right!, it was a clear case of over-engineering.

Fixed, now `hasher` is on the `noder` package.